test/test_unit.py

```diff
@@ -1176,12 +1176,14 @@ def testFindSequenceOnDisk(self):
             ("seq/foo.#.exr", "seq/foo.1-5#.exr"),
             ("seq/foo.#.jpg", "seq/foo.1-5#.jpg"),
             ("seq/foo.0002.jpg", "seq/foo.1-5#.jpg"),
-            ("seq/foo.#.exr", "seq/foo.1-5#.exr"),
             ("seq/foo.debug.#.exr", "seq/foo.debug.1-5#.exr"),
             ("seq/#.exr", "seq/1-3#.exr"),
             ("seq/bar1001.exr", "seq/bar1001.exr"),
             ("seq/foo_0001.exr", "seq/foo_0001.exr"),
             ("multi_range/file_#.0001.exr", "multi_range/file_3-5#.0001.exr"),
+            ("subframe_seq/baz.#.0000.exr", "subframe_seq/baz.1-4#.0000.exr"),
+            ("subframe_seq/baz.0001.#.exr", "subframe_seq/baz.0001.0-7500x2500#.exr"),
+            ("subframe_seq/baz.0001.0000.exr", "subframe_seq/baz.0001.0-7500x2500#.exr"),
```

Comment:
What would happen if we added a test case like:
```
("subframe_seq/baz.#.exr", ?),
```
This is exercising  `allow_subframes=False` so I am assuming the result is that it would not find any sequence? I think there is some code in their that checks if the frame is `Decimal` but I couldn't remember if the logic leading up to that allows it to parse a float "0001.0000"

Reply:
Yeah, it doesn't find the sequence and raises FileSeqException. It's a reasonable test to add though.

findSequenceOnDisk creates a FileSequence from the provided pattern, and uses the dirname/basename/extension to glob the files. The result is filtered by decimal_places (which removes all files in the test above) before being passed to yield_sequences_in_list.

I'll update PR with extra tests to prevent possible regressions in future.

Reply:
Extra test case added